services/rfq/relayer/limiter/limiter.go

```diff
@@ -0,0 +1,116 @@
+// Package limiter provides a rate limiting mechanism for RFQs to protect against reorgs.
+package limiter
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/synapsecns/sanguine/core/metrics"
+	"github.com/synapsecns/sanguine/ethergo/client"
+	"github.com/synapsecns/sanguine/services/rfq/relayer/quoter"
+	"github.com/synapsecns/sanguine/services/rfq/relayer/relconfig"
+	"github.com/synapsecns/sanguine/services/rfq/relayer/reldb"
+	"go.uber.org/ratelimit"
+)
+
+// Limiter is the interface for rate limiting RFQs.
+type Limiter interface {
+	// IsAllowed returns true if the request is allowed, false otherwise.
+	IsAllowed(ctx context.Context, request reldb.QuoteRequest) (bool, error)
+	// Take blocks until a token is available.
+	Take() time.Time
+}
+
+type limiterImpl struct {
+	// TODO: Possibly unneeded?
+	ratelimit.Limiter
+
+	client     client.EVM
+	quoter     quoter.Quoter
+	cfg        relconfig.Config
+	tokenNames map[string]relconfig.TokenConfig
+}
+
+// NewRateLimiter creates a new Limiter.
+// TODO: implement the sliding window: queue up requests and process them in order if cumulative volume is above limit.
+func NewRateLimiter(
+	cfg relconfig.Config,
+	q quoter.Quoter,
+	client client.EVM,
+	metricHandler metrics.Handler,
```

Comment:
Let's definitely add some traces to functions here, particularly when we make the final `IsAllowed()` decision